src/MissionManager/MissionController.cc

```diff
@@ -1886,9 +1884,10 @@ void MissionController::_initVisualItem(VisualMissionItem* visualItem)
     connect(visualItem, &VisualMissionItem::specifiedVehicleYawChanged,                 this, &MissionController::_recalcMissionFlightStatusSignal, Qt::QueuedConnection);
     connect(visualItem, &VisualMissionItem::terrainAltitudeChanged,                     this, &MissionController::_recalcMissionFlightStatusSignal, Qt::QueuedConnection);
     connect(visualItem, &VisualMissionItem::additionalTimeDelayChanged,                 this, &MissionController::_recalcMissionFlightStatusSignal, Qt::QueuedConnection);
-
     connect(visualItem, &VisualMissionItem::lastSequenceNumberChanged,                  this, &MissionController::_recalcSequence);
 
+    visualItem->setMissionFlightStatus(_missionFlightStatus);
```

Comment:
I have moved this here as it wasn't invoked for the complex items and rendered `TransectStyleComplexItem::_missionFlightStatus` uninitialised and in turn [this code](https://github.com/mavlink/qgroundcontrol/blob/master/src/MissionManager/TransectStyleComplexItem.cc#L712) didn't do anything useful.

Reply:
So the fact that setMissionFlightStatus is called in the insert workers likely incorrect. I'd have to go back to see why they were added. The problem is that there are values in _missionFlightStatus which are dynamic and are specific to the point in the recalc sequence. For example vehicle speeds can be adjusted by mission items. So they are tracked as the recalc is done and the current status is stuffed into the items.

Also the problem you are seeing with TransectStyleComplexItem actually indicates a different bug. Which is that if the vehicle speed given to TransectStyleComplesItem changes then terrain follow must be fully rebuilt. Since the vehicle speed affects the generation of interstitial waypoints. Which if you look here:
```
void TransectStyleComplexItem::setMissionFlightStatus(MissionController::MissionFlightStatus_t& missionFlightStatus)
{
    ComplexMissionItem::setMissionFlightStatus(missionFlightStatus);
    if (!qFuzzyCompare(_cruiseSpeed, missionFlightStatus.vehicleSpeed)) {
        _cruiseSpeed = missionFlightStatus.vehicleSpeed;
        emit timeBetweenShotsChanged();
    }
}
```

You'll see that isn't happening.

I'll take a look at figuring what is they correct way to deal with is. Until then this pull will have to wait those fixes. And you should take out you changed for setMissionFlightStatus so my changes when they are ready will merge in.

Reply:
done: https://github.com/mavlink/qgroundcontrol/pull/8846/commits/f38ab2610945bd8dd9a8cc7920369973337f66f8